tools/tplg_parser/tplg_parser.c

```diff
@@ -198,7 +198,7 @@ int tplg_load_pcm(int comp_id, int pipeline_id, int size, int dir,
 	array = (struct snd_soc_tplg_vendor_array *)malloc(size);
 	if (!array) {
 		fprintf(stderr, "error: mem alloc\n");
-		return -EINVAL;
```

Comment:
this should be -ENOMEM and some of the others for malloc too

Reply:
No, malloc() sets errno to appropriate value like ENOMEM. You can see the possible values with command `errno -ls`.